File: typescript/Creational/03-Builder/Director/JobCardDirector.ts
import { JobCard } from "../Product/JobCard";
import { JobCardBuilder } from "../Concrete Builder/JobCardBuilder";

export class JobCardDirector {
  buildEmergencyJob(title: string, category: string): JobCard {
    return new JobCardBuilder(title, category)
      .withPriority("URGENT")
      .requiresSafetyInspection()
      .withInstructions("Emergency callout — contact homeowner immediately on arrival.")
      .build();
  }

  buildScheduledJob(title: string, category: string, slot: Date): JobCard {
    return new JobCardBuilder(title, category)
      .withPriority("NORMAL")
      .scheduledFor(slot)
      .build();
  }
}

File: csharp/Creational/03-Builder/Director/JobCardDirector.cs
public class JobCardDirector
{
    private readonly IJobCardBuilder _builder;

    public JobCardDirector(IJobCardBuilder builder)
    {
        _builder = builder;
    }

    // Pre-set recipe: emergency callout
    public JobCard BuildEmergencyJob(string title, string category) =>
        new JobCardBuilder(title, category)
            .WithPriority("URGENT")
            .RequiresSafetyInspection()
            .WithInstructions("Emergency callout — contact homeowner immediately on arrival.")
            .Build();

    // Pre-set recipe: standard scheduled job
    public JobCard BuildScheduledJob(string title, string category, DateTime slot) =>
        new JobCardBuilder(title, category)
            .WithPriority("NORMAL")
            .ScheduledFor(slot)
            .Build();
}